Develop_profiles-controller＆edit view

FUM01ichikawa opened the pull request:
# WHAT
・プロフィールコントローラーの作成
（・profiles_controller
→Mypageフォルダに格納
→1行目を下記に修正
class Mypage::ProfilesController < ApplicationController）
・プロフィール編集ページのViewの一部作成

# WHY
・ユーザのプロフィール確認及び変更機能の実装に向け

作成画面
https://gyazo.com/a8ddbec9918356f9eda2205a46504870

予定の完成形はメルカリの↓ページ
https://www.mercari.com/jp/mypage/profile/

KassemberKiyoshi committed:
Develop_profiles-controller＆edit view

# WHAT
・プロフィールコントローラーの作成
（・profiles_controller
→Mypageフォルダに格納
→1行目を下記に修正
class Mypage::ProfilesController < ApplicationController）
・プロフィール編集ページのViewの一部作成

# WHY
・ユーザのプロフィール確認及び変更機能の実装に向け

wheesnoza commented on app/views/mypage/profiles/_profile-edit.html.haml:
```diff
@@ -0,0 +1,11 @@
+.container
```
edit.html.erbというファイルを作成してそこにここの内容をコピペしてください！
またこのファイルの名前をedit.html.hamlにしてください！
不明な点があれば渡辺まで。

FUM01ichikawa replied:
ファイル名称変更しました〜

KassemberKiyoshi committed:
ページのファイル名称を変更しました

wheesnoza commented:
LGTM!

i-date commented on app/assets/stylesheets/profiles.scss:
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the profiles controller here.
```
こちらのファイルは不要でしたら削除しましょう。

FUM01ichikawa replied:
ファイルを削除しました

i-date commented on app/views/mypage/profiles/edit.html.haml:
```diff
@@ -0,0 +1,11 @@
+.container
```
こちらのページではフォームを使用しているので、どこかでformタグなどを使用してあげましょう。

FUM01ichikawa replied:
form_tagを追記しました。（内容は今後追記する形になります。）

i-date commented on app/assets/javascripts/profiles.coffee:
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
こちらのファイルは不要でしたら削除しましょう。

FUM01ichikawa replied:
ファイルを削除しました。

i-date reviewed (commented):
ご提出ありがとうございます！コメントいたしました^^

KassemberKiyoshi committed:
レビューいただいた内容を対応しました

・edit.html.hamlにform_tagを追記
・_profile-edit.scssの一部微調整
・指摘いただいた不要２ファイルを削除

FUM01ichikawa commented:
念の為、調整版は下記の形になります。（アイコンとname部分の距離を微調整）
https://gyazo.com/23455ec87062f30dd204b01703382e32

KassemberKiyoshi committed:
profiles_controller_test.rb部分追記

・削除のアドバイスをいただきましたが、追記して残す方向でいきたいです。

FUM01ichikawa commented on test/controllers/profiles_controller_test.rb:
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
testファイルに追記しました。残す方向で進めたいです。

ken-nag replied:
上記どうでしょうか、rspecをお勧めします

KassemberKiyoshi committed:
レビューいただいた内容の変更

・コンフリクトの解消
・背景のURL→「/」が一つ多かったので消しました。
（///→//に自動変換されて画像自体は表示されていました）
・helperのファイルは削除しました

FUM01ichikawa committed:
Merge branch 'master' into Develop_profiles-controller

KassemberKiyoshi committed:
プロフィール画像について対応

アドバイスいただいた通り、該当画像をassets/images以下に格納して呼び出す形にしました！

FUM01ichikawa reviewed (commented):
確認のご指示をいただいた点はこちらで間違いないでしょうか？
（因みにgithub上のUpload filesボタンからも上げる方法もやってみたのですが、github desktopから上げたものと表示は同じになりました。）

FUM01ichikawa commented:
えーと、すいません！改めて確認ですが、image_tagで呼び出している7行目の部分（プロフィール画像）についての変更指示という認識でし対応していましたが、8行目〜11行目の中で変更した方が良い部分（image_tagで別途呼びだす必要がある点）がありましたら助言いただきたいです。
また、↓では画像がアップロードされていない状態でしょうか？
https://gyazo.com/f0f8e1b973ee2dcdf633d44a5033472a

Commit:
Refact shared stylesheet

Commit:
Add shared colors

FUM01ichikawa commented on app/assets/stylesheets/modules/_profile-edit.scss:
```diff
@@ -0,0 +1,75 @@
+.container{
+  width:700px;
+  height:568px;
+  background-color:#FFFFFF;
```
ありがとうございます。カラーに関してはチームで相談して変数化を進めたいと思います。
（一旦Viewの完成を進めています。）

FUM01ichikawa commented on app/views/mypage/profiles/edit.html.haml:
```diff
@@ -0,0 +1,12 @@
+      %input.input-default{name: "nickname", placeholder: "例)AYA☆セール中", type: "text", value: "Fumiaki  Ichikawa"}/
+      .setting-profile-content
+        %textarea.textarea-default{name: "introduction", placeholder: "例）こんにちは☆ ご覧いただきありがとうございます！かわいいものやきれいめオフィスカジュアル中心に出品しています。服のサイズは主に9号です。気持ちよくお取引できるよう心がけていますので、商品や発送方法などご質問がありましたらお気軽にどうぞ♪"}
+        %input{name: "after_save_callback", type: "hidden", value: "https://www.mercari.com/jp/mypage/profile/"}/
```
7行目のimage_tag部分の画像アップロードですが、git上からのアップロードで下記の通り別のpull requestを作成してみました。↓こちらをマージすると、アップロードされるでしょうか（一応Files changedには表示されています。）
https://github.com/YukiTokieda/freemarket_sample_50c/pull/35

ken-nag replied:
確認しました、こちらのprでは画像上がってますね！

ken-nag commented:
テスト方針をrspecにするのかrails標準のものにするのか回答お願いします！

FUM01ichikawa commented:
テスト方針はrspecの方向です。（実際の導入は別のプルリクエストになるかと思います。）

KassemberKiyoshi committed:
Application.scss更新

KassemberKiyoshi committed:
Merge branch 'master' into Develop_profiles-controller

# Conflicts:
#	app/assets/stylesheets/application.scss
#	app/assets/stylesheets/config/colors.scss

TOMARUMARU commented on app/assets/stylesheets/application.scss:
```diff
@@ -5,4 +5,8 @@
 @import 'products';
 @import 'shared/*';
 @import 'modules/*';
+<<<<<<< HEAD
```
conflct解消しましょう

FUM01ichikawa replied:
解消しました。

KassemberKiyoshi committed:
ご指摘の点解消

TOMARUMARU reviewed (approved):
LGTM
